inception-scheduling/src/main/java/de/tudarmstadt/ukp/inception/scheduling/IntrospectiveThreadPoolExecutor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.scheduling;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
+
+public class IntrospectiveThreadPoolExecutor
```

Comment:
What does "introspective" mean here?

Reply:
It means that you can look into what the executor does.

Reply:
I think that would be "Inspectable" then. "Introspective" sounds like the executor looks inside itself.